platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -369,7 +375,73 @@ def __init__(self):
         self.calibration = 0
         self.qsfp_page3_available = False
         self.second_application_list = False
-        
+
+class MlxregManager:
+    def __init__(self, mst_pci_device, slot_id, sdk_index):
+        self.mst_pci_device = mst_pci_device
+        self.slot_id = slot_id
+        self.sdk_index = sdk_index
+
+    def construct_dword(self, write_buffer):
+        res = ""
+        used_bytes_in_dword = len(write_buffer) % BYTES_IN_DWORD
+
+        for idx, x in enumerate(write_buffer):
+            word = hex(x)[2:]
+            if idx == 0:
+                res = "dword[0]=0x"
+            elif idx % BYTES_IN_DWORD == 0:
+                res += "dword[" + str((idx + 1)//BYTES_IN_DWORD) + "]=0x"
+
+            res += word.zfill(2)
+            idx += 1
+            if idx % BYTES_IN_DWORD == 0:
+                res += ","
+
+        if used_bytes_in_dword % BYTES_IN_DWORD != 0:
+            res += (BYTES_IN_DWORD - used_bytes_in_dword) * "00"
+        return res
+
+    def write_mlxreg_eeprom(self, num_bytes, dword, device_address, page):
+        try:
+            cmd = "mlxreg -d /dev/mst/{} --reg_name MCIA --indexes \
+                    slot_index={},module={},device_address={},page_number={},i2c_device_address=0x50,size={},bank_number=0 \
+                    --set {} -y".format(self.mst_pci_device, self.slot_id, self.sdk_index, device_address, page, num_bytes, dword)
+            result = subprocess.check_output(cmd, universal_newlines=True, shell=True)
+
+        except subprocess.CalledProcessError as e:
+            print("Error! Unable to write data for {} port, page {} offset {}, rc = {}, err msg: {}".format(self.sdk_index, page, device_address, e.returncode, e.output))
+            return False
+        return True
+
+    def read_mlxred_eeprom(self, offset, page, num_bytes):
+        try:
+            cmd = "mlxreg -d /dev/mst/{} --reg_name MCIA --indexes \
+                    slot_index={},module={},device_address={},page_number={},i2c_device_address=0x50,size={},bank_number=0 \
+                    --get | grep dword".format(self.mst_pci_device, self.slot_id, self.sdk_index, offset, page, num_bytes)
```

Comment:
@ayurkiv-nvda let's keep the things simple and have parsing done purely in `python`